src/com/trollworks/gcs/feature/SpellBonus.java

```diff
@@ -95,27 +102,40 @@ public String getXMLTag() {
     @Override
     public String getKey() {
         StringBuffer buffer = new StringBuffer();
-
-        if (mAllColleges) {
-            buffer.append(Spell.ID_COLLEGE);
+        if (!mCategoryCriteria.isTypeAnything()) {
```

Comment:
Have you validated that this rearrangement hasn't broken other things?

Reply:
I believe so.   The idea being add this bonus into the feature map, you want to be able to look it up based on the correct criteria.   If the "categories" criteria is NOT "anything", then the user is looking to match this SpellBonus based on something more than just college or one of the various types of "name" matches (college, power source, name).  So instead of putting it in the "All College" or exact "Name" matching list, I put it in the "Name*" list, which is the catch-all for bonuses that have more than one criteria (this is how it works for Skill and Weapon Bonuses).

It seems to work, and I have tried a few different combinations of spell bonus matching but I will try more (all, if I can work out all of the combinations) before I mark this resolved.

Reply:
Ok, that took less time than I thought.   I created a bunch of test advantages and spells:
![image](https://user-images.githubusercontent.com/8931036/60153764-455a5980-97b3-11e9-8575-0be35ad17f5d.png)
The advantages described what "name match" they were looking for (Colleges, Spell Name, Power) and which category.   The Spells were created with the various "name"s as Test (College, Spell Name, Power), and then a duplicate version with the category set to Test.  

I verified that each spell got the appropriate additions from each advantage:
Test
![image](https://user-images.githubusercontent.com/8931036/60153841-91a59980-97b3-11e9-8d1e-de2185cf28ab.png)

Test Cat: Test
![image](https://user-images.githubusercontent.com/8931036/60154028-14c6ef80-97b4-11e9-9ac8-2a992a1f3fb4.png)

College:Test
![image](https://user-images.githubusercontent.com/8931036/60154056-33c58180-97b4-11e9-8024-4691bf341695.png)

College:Test Cat:Test
![image](https://user-images.githubusercontent.com/8931036/60154119-64a5b680-97b4-11e9-999a-fba45d3a71ce.png)

Power: Test
![image](https://user-images.githubusercontent.com/8931036/60154195-7dae6780-97b4-11e9-8352-f916bb0b5293.png)

Power: Test Cat:Test
![image](https://user-images.githubusercontent.com/8931036/60154213-9585eb80-97b4-11e9-9230-121bd5298624.png)

I believe that I have covered all of the combinations offered by the Feature editor.